gammapy/irf/effective_area_table.py

```diff
@@ -444,6 +465,44 @@ def read(filename):
         hdu_list = fits.open(filename)
         return EffectiveAreaTable2D.from_fits(hdu_list)
 
+    def to_effective_area_table(self, offset, energ_lo=None, energ_hi=None):
+        """Evaluate at a given offset and return effective area table instance.
+
+        The energy thresholds in the effective area table object are not set.
+        If the effective area table is intended to be used for spectral analysis,
+        the final energy binning should be given here, since the
+        effective area table class does no interpolation.
+
+        Parameters
+        ----------
+        offset : `~astropy.coordinates.Angle`
+            offset
+        energy : `~astropy.units.Quantity`
+            energy
+
+        Returns
+        -------
+        eff_area_table : `EffectiveAreaTable`
+             Effective area table instance. 
+        """
+
+        if energ_lo is None and energ_lo is None:
+            energ_lo = self.energ_lo
+            energ_hi = self.energ_hi
+        elif energ_lo is None or energ_hi is None:
+            raise ValueError("Only 1 energy vector given, need 2")
+        if not isinstance(offset, Angle):
```

Comment:
Move this offset input validation below all the energy input checks (and separate by a blank line)